modules/cli/src/main/java/org/apache/ignite/internal/cli/call/recovery/PartitionStatesCall.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.cli.call.recovery;
+
+import static java.util.stream.Collectors.toList;
+
+import jakarta.inject.Singleton;
+import java.util.List;
+import java.util.stream.Stream;
+import org.apache.ignite.internal.cli.core.call.Call;
+import org.apache.ignite.internal.cli.core.call.DefaultCallOutput;
+import org.apache.ignite.internal.cli.core.exception.IgniteCliApiException;
+import org.apache.ignite.internal.cli.core.rest.ApiClientFactory;
+import org.apache.ignite.internal.cli.sql.table.Table;
+import org.apache.ignite.rest.client.api.RecoveryApi;
+import org.apache.ignite.rest.client.invoker.ApiException;
+import org.apache.ignite.rest.client.model.GlobalPartitionStatesResponse;
+import org.apache.ignite.rest.client.model.LocalPartitionStatesResponse;
+
+/** Call to get partition states. */
+@Singleton
+public class PartitionStatesCall implements Call<PartitionStatesCallInput, Table> {
+    private final ApiClientFactory clientFactory;
+
+    private static final List<String> GLOBAL_HEADERS = List.of("Zone name", "Table name", "Partition ID", "State");
+
+    private static final List<String> LOCAL_HEADERS = Stream
+            .concat(Stream.of("Node name"), GLOBAL_HEADERS.stream())
+            .collect(toList());
+
+    public PartitionStatesCall(ApiClientFactory clientFactory) {
+        this.clientFactory = clientFactory;
+    }
+
+    @Override
+    public DefaultCallOutput<Table> execute(PartitionStatesCallInput input) {
+        RecoveryApi client = new RecoveryApi(clientFactory.getClient(input.clusterUrl()));
+
+        List<String> trimmedZoneNames = trim(input.zoneNames());
+        List<String> trimmedNodeNames = trim(input.nodeNames());
+
+        try {
+            if (input.local()) {
+                LocalPartitionStatesResponse localStates = client.getLocalPartitionStates(
```

Comment:
I suggest extracting both branches of this `if` as methods: one for handling local case, another for handling global case